kong/plugins/basic-auth/basicauth_credentials.lua

```diff
@@ -1,5 +1,20 @@
 local crypto = require "kong.plugins.basic-auth.crypto"
-local utils  = require "kong.tools.utils"
+
+
+local type = type
+local re_find = ngx.re.find
+
+
+local uuid_regex = "^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$"
+
+
+local function is_valid_uuid(str)
+  if type(str) ~= 'string' or #str ~= 36 then
+    return false
+  end
+
+  return re_find(str, uuid_regex, 'ioj') ~= nil
+end
```

Comment:
Why are we duplicating `is_valid_uuid`? This is a blocker. The scope of this work is not to get rid of `kong.tools.utils`.